src/main/java/pl/iordervivi/data/scrapper/gov/covid19/service/SickService.java

```diff
@@ -0,0 +1,62 @@
+package pl.iordervivi.data.scrapper.gov.covid19.service;
+
+import org.springframework.stereotype.Service;
+import pl.iordervivi.data.scrapper.gov.covid19.domain.Region;
+import pl.iordervivi.data.scrapper.gov.covid19.domain.Sick;
+import pl.iordervivi.data.scrapper.gov.covid19.dto.SickInRegionsDto;
+import pl.iordervivi.data.scrapper.gov.covid19.repo.RegionRepository;
+import pl.iordervivi.data.scrapper.gov.covid19.repo.SickRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class SickService {
+    private final SickRepository sickRepository;
+    private final RegionRepository regionRepository;
+
+    public SickService(SickRepository sickRepository, RegionRepository regionRepository) {
+        this.sickRepository = sickRepository;
+        this.regionRepository = regionRepository;
+    }
+
+    void addSick(List<SickInRegionsDto> sickInRegionsDtos, LocalDateTime now, long step) {
+        long sick;
+        long death;
+        for (SickInRegionsDto sickInRegionsDTO : sickInRegionsDtos) {
+            sick = 0;
+            death = 0;
+            Optional<Region> region = regionRepository.findByRegion(sickInRegionsDTO.getRegionName());
+            if (region.isEmpty()) {
+                String er = "Region not found in DB during adding sick data to the database: " + sickInRegionsDTO.getRegionName();
+                throw new NullPointerException(er);
+            }
+            if (sickInRegionsDTO.getTotalSickInRegion() != null) {
```

Comment:
cała logika zerowania powinna być uproszczona do:
`long sick =Optional.ofNullable(sickInRegionsDTO.getTotalSickInRegion()).orElse(0);`

A jak w klasie `SickInRegionsDTO` przypiszesz domyślnie 0 to tutaj możesz odrazu używać `.getTotalSickInRegion()`

Reply:
dzięki za radę, rozumiem że tak naprawdę wystarczyło to zrobić żeby użyć dalej `SickInRegionsDTO` 
```     
@JsonProperty("sick")
    private Long totalDiseaseCasesInRegion = 0L;

    @JsonProperty("deaths")
    private Long totalDeathCasesInRegion = 0L;
```